### PR TITLE
Module: Split it up

### PR DESCRIPTION
**Pull request by gtk-grafana**
Fixes: https://github.com/grafana/explore-logs/issues/764

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -73,6 +73,7 @@
     "tsconfig-paths": "^4.2.0",
     "typescript": "4.8.4",
     "webpack": "^5.86.0",
+    "webpack-bundle-analyzer": "^4.10.2",
```
Comment (matyax) [inline package.json:76]: Are we keeping this one?

**Review by matyax — APPROVED** [review]
Awesome!

File: webpack.config.ts [hunk 2]
```diff
@@ -1,14 +1,18 @@
 import type { Configuration } from 'webpack';
```
Comment (matyax) [inline webpack.config.ts:1]: Same, just double checking we want to keep these changes.

File: src/module.tsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import { lazy } from 'react';
+import { AppPlugin } from '@grafana/data';
+
+const App = lazy(async () => {
```
Comment (svennergr) [inline src/module.tsx:4]: Hm, I think the `lazy` breaks preloading, thus links from Explore and panels don't work with this PR.

**Review by svennergr — CHANGES_REQUESTED** [review]

**Comment by gtk-grafana** [comment]
Breaks extensions, apparently there's no way to register links async, we can either break stuff up as much as possible to make the module bundle just the links and the lezer/logql package:
https://github.com/grafana/explore-logs/pull/768

Or we'll need to rewrite the parsing to exclude lezer. Either way this will have to be dealt with post GA

**Pull request closed without merging**